mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,819 @@
+                                    continue;
+                                }
+
+                                let fee_details = fee_details.clone();
+                                let mut fee_tx_details = details;
+                                fee_tx_details.to = vec![];
+                                fee_tx_details.total_amount = fee_details.amount.clone();
+                                fee_tx_details.spent_by_me = fee_details.amount.clone();
+                                fee_tx_details.received_by_me = BigDecimal::default();
+                                fee_tx_details.my_balance_change = BigDecimal::default() - &fee_details.amount;
+                                fee_tx_details.fee_details = None;
+                                fee_tx_details.coin = coin.platform_ticker().to_string();
+                                // Non-reversed version of original internal id
+                                fee_tx_details.internal_id = H256::from(internal_id_hash.as_slice()).to_vec().into();
+                                fee_tx_details.transaction_type = TransactionType::Fee(token_id);
+
+                                tx_details.push(fee_tx_details);
+                                fee_added = true;
+                            }
+                        }
+                    }
+
+                    log::debug!("Tx '{}' successfuly parsed.", tx.hash);
+                }
+
+                try_or_return_stopped_as_err!(
+                    storage
+                        .add_transactions_to_history(&coin.history_wallet_id(), tx_details)
+                        .await
+                        .map_err(|e| format!("{:?}", e)),
+                    StopReason::StorageError,
+                    "add_transactions_to_history failed"
+                );
+
+                iterate_more = (TX_PAGE_SIZE as u32 * page) < response.total_count;
+                page += 1;
+            }
+
+            Ok(highest_height)
+        }
+
+        let q = format!(
+            "coin_spent.spender = '{}' AND tx.height > {}",
+            self.address.clone(),
+            self.from_block_height
+        );
+        let highest_send_tx_height = match fetch_and_insert_txs(
+            self.address.clone(),
+            &ctx.coin,
+            &ctx.storage,
+            q,
+            self.from_block_height,
+        )
+        .await
+        {
+            Ok(block) => block,
+            Err(stopped) => {
+                if let StopReason::RpcClient(e) = &stopped.stop_reason {
+                    log::error!("Sent tx history process turned into cooldown mode due to rpc error: {e}");
+                    return Self::change_state(OnIoErrorCooldown::new(self.address.clone(), self.from_block_height));
+                }
+
+                return Self::change_state(stopped);
+            },
+        };
+
+        let q = format!(
+            "coin_received.receiver = '{}' AND tx.height > {}",
```

Comment:
Yes this can work but I don't see the point why we are excluding self-send txs and then re-requesting for it. Also, for some txs like htlcs 3rd query wouldn't work. They have some weird magical work with different addresses. If you send htlc to some address, IRIS sending it to their own address and then controlling the htlc flow.

Reply:
Ok! I guess it's not a common case to send funds to yourself, so there shouldn't be many self-sent transactions anyway